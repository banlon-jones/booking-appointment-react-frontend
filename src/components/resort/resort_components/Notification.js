/* eslint-disable prefer-const */
import React from 'react';
import AddResort from '../AddResort.module.css';

function Notification(props) {
  // eslint-disable-next-line react/prop-types
  let { showNotice, message } = props;

  return (
    <>
      <div
        className={AddResort.notice_container}
        style={{
          display: message && showNotice === true ? 'block' : 'none',
        }}
      >
        {message && (
          <p
            className={AddResort.notification}
            style={{
              backgroundColor:
                message === 'Could not update resort' ? 'red' : ' ',
            }}
          >
            {message}
          </p>
        )}
      </div>
    </>
  );
}

export default Notification;
