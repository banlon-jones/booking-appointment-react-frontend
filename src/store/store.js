/* eslint-disable import/no-named-as-default */
/* eslint-disable import/no-named-as-default-member */
import { configureStore } from '@reduxjs/toolkit';
import userSlice from './user/user';
import resortDetailSlice from './resort-details/resortDetail';

const store = configureStore({
  reducer: {
    user: userSlice,
    resortDetail: resortDetailSlice,
  },
});

export default store;
