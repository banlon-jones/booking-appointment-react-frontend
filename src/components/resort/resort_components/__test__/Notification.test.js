/* eslint-disable comma-dangle */
import React from 'react';
import { cleanup, screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../../store/store';
import Notification from '../Notification';

afterEach(cleanup);

describe('React Test for AddResort', () => {
  test('Notification Test', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Notification />
        </Provider>
      </BrowserRouter>
    );
    const ResortElement = screen.getByTestId('notice');
    expect(ResortElement).toBeInTheDocument('resort');
  });
});

describe('Notification Snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Notification />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
