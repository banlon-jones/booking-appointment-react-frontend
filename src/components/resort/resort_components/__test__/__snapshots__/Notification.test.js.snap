// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Snapshot renders correctly 1`] = `
<div
  className="notice_container"
  data-testid="notice"
  style={
    Object {
      "display": "none",
    }
  }
/>
`;
